source_gen/lib/builder.dart

```diff
@@ -78,8 +86,13 @@ class CombiningBuilder implements Builder {
     });
 
     var assets = await new Stream.fromIterable(assetIds)
-        .asyncMap(buildStep.readAsString)
-        .map((s) => s.trim())
+        .asyncMap((id) async {
+          var content = (await buildStep.readAsString(id)).trim();
+          if (_includePartName) {
+            content = '// Part: ${id.pathSegments.last}\n$content';
```

Comment:
do we want to omit empty parts?

Reply:
Nope. This is intentional. Helps folks see that there was a part – just the content was empty.